src/utils/beacon/decodePayload.ts

```diff
@@ -0,0 +1,52 @@
+import { hex2buf } from "@taquito/utils";
+
+const PAYLOAD_PADDING = "0501";
+
+/**
+ * Decodes a sign request payload string.
+ *
+ * @param payload - The payload to decode.
+ * @returns The decoded string, or the original payload if decoding fails.
+ * @see {@link https://taquito.io/docs/signing/} for more info.
+ */
+export const decodePayload = (payload: string): string => {
```

Comment:
you could also change the type of payload to `SignPayloadRequestOutput['payload']`

that way, you can clearly see and, what's more important, navigate to the actual type definition